src/ObjCRuntime/Blocks.cs

```diff
@@ -83,14 +83,19 @@ public unsafe struct BlockLiteral {
 		[DllImport ("__Internal")]
 		static extern IntPtr xamarin_get_block_descriptor ();
 
+		struct DelegatePairs {
+			public Delegate Trampoline, UserDelegate;
+		}
+
 		//
 		// trampoline must be static, and someone else needs to keep a ref to it
 		//
 		public unsafe void SetupBlock (Delegate trampoline, Delegate userDelegate)
 		{
 			isa = block_class;
 			invoke = Marshal.GetFunctionPointerForDelegate (trampoline);
-			local_handle = (IntPtr) GCHandle.Alloc (userDelegate);
+			var delegates = new DelegatePairs () { Trampoline = trampoline, UserDelegate = userDelegate };
```

Comment:
Irrespective of the documentation lowering the requirements of the _public_ API seems a good idea, it's too easy to misuse.

@rolfbjarne any side effects you wanted to avoid ? beside requiring a bit more memory
or is it about our generated code ? which already satisfy this requirement automatically